sql/path.cc

```diff
@@ -760,18 +727,17 @@ initStringUnchecked(T && str)
 {
     // This method is used only for when we know we may have invalid
     // characters, for example when importing legacy files.
-    words[0] = words[1] = words[2] = 0;
-    digits_ = calcDigits(rawData(str), rawLength(str));
-    if (rawLength(str) > 0 && rawLength(str) <= INTERNAL_BYTES - 1) {
-        complex_ = 0;
-        simpleLen_ = rawLength(str);
-        std::copy(rawData(str), rawData(str) + rawLength(str),
-                  bytes + 1);
+    const char * d = rawData(str);
+    size_t l = rawLength(str);
+    digits_ = calcDigits(d, l);
+    if (l == 0) {
+        storage_.append("\0", 1);
     }
     else {
-        complex_ = 1;
-        new (&this->str.str) Utf8String(std::move(str));
+        storage_.append(d, l);
     }
+
+    //ExcAssertEqual(toUtf8String(), str);
```

Comment:
rm if unecessary

Reply:
For me it's helpful documentation.  This is a postcondition, that is too expensive to check, but is helpful to understand what is going on.